assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,199 @@
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			.map( ( { answer_ordinal, answer_text } ) => {
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			if ( write_in ) {
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < min_choices;
+	const hasMoreThanMaxChoices =
+		max_choices && totalSelectedValues > max_choices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{ choices.map( ( { answer_ordinal, text, write_in } ) => {
+					const answer = selectedValues[ answer_ordinal ];
+
+					return (
+						<div key={ text } style={ { display: 'flex' } }>
+							<Checkbox
+								checked={ answer.selected }
+								onChange={ () => handleCheck( answer_ordinal ) }
+								value={ `${ answer_ordinal }` }
+								id={ text }
+								name={ text }
+							>
+								{ text }
+							</Checkbox>
+							{ write_in && (
+								<TextField>
+									<Input
+										onChange={ ( event ) =>
+											handleAnswerChange(
+												event,
+												answer_ordinal
+											)
+										}
+										value={ answer.answer_text }
+										disabled={ ! answer.selected }
+										aria-label={ `Text input for option ${ text }` }
+									/>
+								</TextField>
+							) }
+						</div>
+					);
+				} ) }
+			</div>
+
+			<div className="googlesitekit-survey__footer">
+				<Button
+					onClick={ handleSubmit }
+					disabled={ isSubmitButtonDisabled }
+				>
+					Next
```

Comment:
This should be localised:
```suggestion
					{ __( 'Next', 'google-site-kit' ) }
```

Reply:
One thing I meant to bring up, should this be configurable with props?

As the design (the screenshot :wink: ) for 3762 (the open text) shows the text _Submit_ (which I've hardcoded)

But the design for this and the Single Select shows _Next_

I would guess that this should be a prop :thinking:

Reply:
Hmm, indeed. I don't know if it needs to be a prop exactly, but that's an implementation detail—I think in this case it would make sense to be one.

But mainly: it makes sense that it should be `Submit` when it's the last question (when the answer ordinal is the same as the total length of questions) and `Next` otherwise.

We don't mention that in the ACs or IBs, but I think it's clear how to do that—in `CurrentSurvey` we should check to see if `questions?.length === currentQuestionOrdinal`. If so, the button should be `Submit`. Otherwise: `Next`.

Reply:
Done in this PR :+1: 

If you're ok with this implementation then we could merge this PR first, then I can update the other PRs